common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java

```diff
@@ -95,6 +111,38 @@ public static String getRemoteAddress(Channel channel) {
     return "<unknown remote>";
   }
 
+  /**
+   * Returns the default number of threads for both the Netty client and server thread pools.
+   * If numUsableCores is 0, we will use Runtime get an approximate number of available cores.
+   */
+  public static int defaultNumThreads(int numUsableCores) {
+    final int availableCores;
```

Comment:
nit: remove `final`

Reply:
Why? 

That final is correct there: it is basically a constant which can get once value and it enforces that before any reading of its value it must be initialised.

Reply:
I think `final` is fine and would use it more in Java code, as we of course use the equivalent `val` by default in Scala.